src/SIM/AdaptiveSIM.C

```diff
@@ -418,6 +420,17 @@ bool AdaptiveSIM::adaptMesh (int iStep, std::streamsize outPrec)
   case MINIMUM: // beta percent of min error (more than 100%)
     limit = errors.back().first  * beta/100.0;
     break;
+  case DORFEL:
+  {
+    for (const DblIdx& error : errors)
+      sumErr += error.first;
+    double total = 0.0;
+    size_t i = 0;
+    while (total < sumErr*beta/100.0)
+      total += errors[i++].first;
+    limit = errors[i].first;
```

Comment:
Cosmetic alternative to lines 428-431:
```
sumErr *= beta/100.0;
for (const DblIdx& error : errors)
  if (total < sumErr)
    total += error.first;
  else
  {
    limit = error.first;
    break;
  }
```
and we don't need the brace-pair after the case-statement, if the `total`variable is declared outside together with `limit`and `sumErr`.

Reply:
i'll do the rewrite, but if i make total non-local i will change name. it's too easily confused with all sorts of things outside the local context.

Reply:
sure, looks good :-) But maybe also initialize limit to zero (or maybe `errors.back().first`then is the correct value) before the loop, just to cover the unlikely event of a very big beta value such that curErr never gets above sumErr

Reply:
done.